packages/client/src/schedule-types.ts

```diff
@@ -0,0 +1,830 @@
+   *
+   * Sorted from older start time to newer.
+   */
+  recentActions: ScheduleAction[];
+
+  /** Next 10 scheduled Action times */
+  nextActionTimes: Date[];
+}
+
+/**
+ * The current Schedule details. They may not match the Schedule as created because:
+ * - some fields in the state are modified automatically
+ * - the schedule may have been modified by {@link ScheduleHandle.update} or
+ *   {@link ScheduleHandle.pause}/{@link ScheduleHandle.unpause}
+ */
+export type Schedule = ListScheduleEntry & {
+  /**
+   * Which Action to take
+   */
+  action: ScheduleActionOptions;
+
+  /**
+   * Controls what happens when an Action would be started by a Schedule at the same time that an older Action is still
+   * running.
+   *
+   * @default {@link ScheduleOverlapPolicy.SKIP}
+   */
+  overlap: ScheduleOverlapPolicy;
+
+  /**
+   * The Temporal Server might be down or unavailable at the time when a Schedule should take an Action. When the Server
+   * comes back up, `catchupWindow` controls which missed Actions should be taken at that point. The default is one
+   * minute, which means that the Schedule attempts to take any Actions that wouldn't be more than one minute late. It
+   * takes those Actions according to the {@link ScheduleOverlapPolicy}. An outage that lasts longer than the Catchup
+   * Window could lead to missed Actions. (But you can always {@link ScheduleHandle.backfill}.)
+   *
+   * @default 1 minute
+   * @format number of milliseconds
+   */
+  catchupWindow: number | undefined;
+
+  /**
+   * When an Action times out or reaches the end of its Retry Policy, {@link pause}.
+   *
+   * With {@link ScheduleOverlapPolicy.ALLOW_ALL}, this pause might not apply to the next Action, because the next Action
+   * might have already started previous to the failed one finishing. Pausing applies only to Actions that are scheduled
+   * to start after the failed one finishes.
+   *
+   * @default false
+   */
+  pauseOnFailure: boolean;
+
+  /**
+   * The Actions remaining in this Schedule. Once this number hits `0`, no further Actions are taken (unless {@link
+   * ScheduleHandle.trigger} is called).
+   *
+   * @default undefined (unlimited)
+   */
+  remainingActions?: number;
+
+  /** Number of Actions taken so far. */
+  numActionsTaken: number;
+
+  /** Number of times a scheduled Action was skipped due to missing the catchup window. */
+  numActionsMissedCatchupWindow: number;
+
+  /** Number of Actions skipped due to overlap. */
+  numActionsSkippedOverlap: number;
+
+  /**
+   * Currently-running workflows started by this schedule. (There might be
+   * more than one if the overlap policy allows overlaps.)
+   */
+  runningWorkflows: WorkflowExecutionWithFirstExecutionRunId[];
+
+  createdAt: Date;
+  lastUpdatedAt: Date | undefined;
+
+  raw: temporal.api.workflowservice.v1.IDescribeScheduleResponse;
+};
+
+export type UpdatedSchedule<Action extends Workflow> = Pick<
```

Comment:
Let’s make this `Omit<ScheduleOptions<Action>, ‘state’>` and make sure that anything that can be updated is not part of the state.